libs/ui/src/components/molecules/GatedInput/GatedInput.tsx

```diff
@@ -0,0 +1,50 @@
+import { ReactNode, useEffect, useState, FunctionComponent } from 'react';
+import { FieldValues, useFormContext } from 'react-hook-form';
+import { CheckedState } from '@radix-ui/react-checkbox';
+import { FieldWrapper } from '../FieldWrapper';
+import { Checkbox } from '../../atoms/Checkbox';
+import { Buildable, IGatedInput } from '../../../types/formAndField';
+import { GatedInputBox } from './GatedInput.styles';
+
+type GatedInputProps = IGatedInput & {
+  gatedId: string;
+  gateLabel: string;
+  input: FunctionComponent<Buildable<IGatedInput>>;
+  updateHelperMsg?: (value: FieldValues) => string; 
+  children?: ReactNode;
+};
+
+export const GatedInput = ({ children, gatedId, gateLabel, input: Input, ...props }: Buildable<GatedInputProps>) => {
+  const { watch } = useFormContext();
+  const { updateHelperMsg } = props;
+  const [gatedOn, toggleGate]= useState(false);
+  const [helperText, setHelperText] = useState(props.helperText || '');
+  const inputValues = watch();
```

Comment:
Can we can scope this to the actual input that input is watching?

const inputValue = watch(id)

Reply:
InputSelect is an example where you might need access to values in both input and select components

Reply:
Better to do const [inputVal, selectVal] = watch([inputId, selectId])